src/XMakeTasks/AssemblyDependency/ResolveAssemblyReference.cs

```diff
@@ -1602,7 +1602,7 @@ private void LogAssembliesConsideredAndRejected(Reference reference, MessageImpo
                                 break;
                             }
                         case NoMatchReason.FusionNamesDidNotMatch:
-                            Log.LogMessageFromResources(importance, "ResolveAssemblyReference.EightSpaceIndent", Log.FormatResourceString("ResolveAssemblyReference.ConsideredAndRejectedBecauseFusionNamesDidntMatch", location.FileNameAttempted, location.AssemblyName.FullName));
+                            Log.LogMessageFromResources(importance, "ResolveAssemblyReference.EightSpaceIndent", Log.FormatResourceString("ResolveAssemblyReference.ConsideredAndRejectedBecauseFusionNamesDidntMatch", location.FileNameAttempted, location.AssemblyName.FullName, location.AssemblyName.Name));
```

Comment:
Call chain: `LogResults` -> `LogReference` -> `LogAssembliesConsideredAndRejected`

Looking at the call sites of LogReference, [1](https://github.com/Microsoft/msbuild/blob/master/src/XMakeTasks/AssemblyDependency/ResolveAssemblyReference.cs#L918) and [2](https://github.com/Microsoft/msbuild/blob/master/src/XMakeTasks/AssemblyDependency/ResolveAssemblyReference.cs#L930), fusionName is being set to `assemblyName.FullName`.

IMHO, irrespective of what `fusionName` is being set to, up the call stack, `LogAssembliesConsideredAndRejected` should use the `fusionName` that it's caller `LogReference` was passed.